pygeos/test/test_constructive.py

```diff
@@ -480,6 +480,99 @@ def test_polygonize_missing():
+        pygeos.Geometry("LINESTRING (1 1, 100 100)"),
+        pygeos.Geometry("POINT (0 0)"),
+        None,
+    ]
+    result = pygeos.polygonize_full(lines)
+    assert len(result) == 4
+    assert all(pygeos.get_type_id(geom) == 7 for geom in result)  # GeometryCollection
+    polygons, cuts, dangles, invalid = result
+    expected_polygons = pygeos.Geometry(
+        "GEOMETRYCOLLECTION (POLYGON ((0 0, 1 1, 1 0, 0 0)), POLYGON ((1 1, 0 0, 0 1, 1 1)))"
+    )
+    assert polygons == expected_polygons
+    assert cuts == pygeos.Geometry("GEOMETRYCOLLECTION EMPTY")
+    expected_dangles = pygeos.Geometry(
+        "GEOMETRYCOLLECTION (LINESTRING (1 1, 100 100), LINESTRING (5 5, 6 6))"
+    )
+    assert dangles == expected_dangles
+    assert invalid == pygeos.Geometry("GEOMETRYCOLLECTION EMPTY")
+
+
+def test_polygonize_full_array():
+    lines = [
+        pygeos.Geometry("LINESTRING (0 0, 1 1)"),
+        pygeos.Geometry("LINESTRING (0 0, 0 1)"),
+        pygeos.Geometry("LINESTRING (0 1, 1 1)"),
+    ]
+    expected = pygeos.Geometry(
+        "GEOMETRYCOLLECTION (POLYGON ((1 1, 0 0, 0 1, 1 1)))"
+    )
+    result = pygeos.polygonize_full(np.array(lines))
+    assert len(result) == 4
+    assert all(isinstance(geom, pygeos.Geometry) for geom in result)
+    assert result[0] == expected
+
+    result = pygeos.polygonize_full(np.array([lines]))
+    assert len(result) == 4
+    assert all(isinstance(geom, np.ndarray) for geom in result)
+    assert all(geom.shape == (1,) for geom in result)
+    assert result[0][0] == expected
+
+    arr = np.array([lines, lines])
+    assert arr.shape == (2, 3)
+    result = pygeos.polygonize_full(arr)
+    assert len(result) == 4
+    assert all(isinstance(arr, np.ndarray) for arr in result)
+    assert all(arr.shape == (2,) for arr in result)
+    assert result[0][0] == expected
+    assert result[0][1] == expected
+
+    arr = np.array([[lines, lines], [lines, lines], [lines, lines]])
+    assert arr.shape == (3, 2, 3)
+    result = pygeos.polygonize_full(arr)
+    assert len(result) == 4
+    assert all(isinstance(arr, np.ndarray) for arr in result)
+    assert all(arr.shape == (3, 2) for arr in result)
+    for res in result[0].flatten():
+        assert res == expected
+
+
+@pytest.mark.skipif(
+    np.__version__ < "1.15",
+    reason="axis keyword for generalized ufunc introduced in np 1.15",
+)
+def test_polygonize_full_array_axis():
+    lines = [
+        pygeos.Geometry("LINESTRING (0 0, 1 1)"),
+        pygeos.Geometry("LINESTRING (0 0, 0 1)"),
+        pygeos.Geometry("LINESTRING (0 1, 1 1)"),
+    ]
+    arr = np.array([lines, lines])  # shape (2, 3)
+    result = pygeos.polygonize_full(arr, axis=1)
+    assert len(result) == 4
+    assert all(arr.shape == (2,) for arr in result)
+    result = pygeos.polygonize_full(arr, axis=0)
+    assert len(result) == 4
+    assert all(arr.shape == (3,) for arr in result)
+
+
+def test_polygonize_full_missing():
```

Comment:
It is probably a good idea to add another test case or cases that include inputs that are mixed valid geometries and `None` to be sure that `None`s are properly ignored.

Reply:
That should already be covered by `test_polygonize_full` (which has a `None` in the input as well), I think?

Reply:
Adding a test with `None` in the first position followed by valid geometries and possibly (though overkill) one in between two valid geometries is probably a good idea; this has helped us catch segfaults in the past (when geometries added to a malloc'd array are first tested for `NULL` before inserting into a position in that array)

Reply:
Yes, adding another `None` to `test_polygonize_full` would be just fine.